performance/performance_tuning.rst

```diff
@@ -241,14 +241,14 @@ There are two configuration parameters which relate to the format in which inter
 
 However, for certain data types like hll or hstore arrays, the cost of serializing and deserializing data is pretty high. In such cases, using binary format for transferring intermediate data can improve query performance due to reduced CPU usage. There are two configuration parameters which can be used to tune this behaviour, citus.binary_master_copy_format and citus.binary_worker_copy_format. Enabling the former uses binary format to transfer intermediate query results from the workers to the coordinator while the latter is useful in queries which require dynamic shuffling of intermediate data between workers.
 
-Real Time Executor
+Adaptive Executor
 -------------------------------
 
-If you have SELECT queries which require sub-second response times, you should try to use the real-time executor.
+If you have SELECT queries which require sub-second response times, you should use the adaptive executor.
 
-The real-time executor opens one connection and uses two file descriptors per unpruned shard (Unrelated shards are pruned away during planning). Due to this, the executor may need to open more connections than max_connections or use more file descriptors than max_files_per_process if the query hits a high number of shards.
+The adaptive executor opens one connection and uses two file descriptors per unpruned shard (Unrelated shards are pruned away during planning). Due to this, the executor may need to open more connections than max_connections or use more file descriptors than max_files_per_process if the query hits a high number of shards.
```

Comment:
I'll re-write this part, so please hold on merge for 1 day

Reply:
I suggest the following (please check the grammer/wording:)


Adaptive Executor
Adaptive executor is the default executor, and it can handle all workloads very efficiently.

The adaptive executor is smart enough to use as less connections as possible to utilize minimal resources on the worker nodes for multi-shard queries. The executor starts using a single connection to a remote node, which is almost always cached so it doesn't incur any additional overhead. If the query execution finishes in `citus.executor_slow_start_interval`, we're done, no more  connections are opened. Else, the executor keeps establishing new connections in every `citus.executor_slow_start_interval`.

With the behaviour explained above, the executor aims to open optimal number of connections to remote nodes. For short running multi-shard queries, like an index-only-scan on the shards, the executor may use only a single connection and execute all the queries on the shards over a single connection. For longer running multi-shard queries, the executor will keep opening connections to parallelize the execution on the remote nodes. If the queries on the shards take long (such as > 500ms), the executor converges to using one connection per shard (or up to `citus.max_adaptive_executor_pool_size`), in order to maximize the parallelism.